server/src/main/java/keywhiz/service/daos/AclDAO.java

```diff
@@ -26,104 +29,119 @@
 import keywhiz.api.model.SecretContent;
 import keywhiz.api.model.SecretSeries;
 import keywhiz.api.model.SecretSeriesAndContent;
-import org.skife.jdbi.v2.sqlobject.Bind;
-import org.skife.jdbi.v2.sqlobject.BindBean;
-import org.skife.jdbi.v2.sqlobject.CreateSqlObject;
-import org.skife.jdbi.v2.sqlobject.SqlQuery;
-import org.skife.jdbi.v2.sqlobject.SqlUpdate;
-import org.skife.jdbi.v2.sqlobject.Transaction;
-import org.skife.jdbi.v2.sqlobject.customizers.RegisterMapper;
-import org.skife.jdbi.v2.sqlobject.customizers.SingleValueResult;
+import org.jooq.DSLContext;
+import org.jooq.Record;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 import static com.google.common.base.Preconditions.checkArgument;
 import static com.google.common.base.Preconditions.checkNotNull;
 import static java.lang.String.format;
-
-@RegisterMapper({SecretSeriesMapper.class, GroupMapper.class, ClientMapper.class})
-public abstract class AclDAO {
+import static keywhiz.jooq.tables.Accessgrants.ACCESSGRANTS;
+import static keywhiz.jooq.tables.Clients.CLIENTS;
+import static keywhiz.jooq.tables.Groups.GROUPS;
+import static keywhiz.jooq.tables.Memberships.MEMBERSHIPS;
+import static keywhiz.jooq.tables.Secrets.SECRETS;
+import static keywhiz.jooq.tables.SecretsContent.SECRETS_CONTENT;
+
+public class AclDAO {
   private static final Logger logger = LoggerFactory.getLogger(AclDAO.class);
 
-  // Creates objects with same underlying connection.
-  @CreateSqlObject protected abstract ClientDAO createClientDAO();
-  @CreateSqlObject protected abstract GroupDAO createGroupDAO();
-  @CreateSqlObject protected abstract SecretContentDAO createSecretContentDAO();
-  @CreateSqlObject protected abstract SecretSeriesDAO createSecretSeriesDAO();
+  private final DSLContext dslContext;
+  private ClientDAO clientDAO;
+  private GroupDAO groupDAO;
+  private SecretContentDAO secretContentDAO;
+  private SecretSeriesDAO secretSeriesDAO;
+
+  @Inject
+  public AclDAO(DSLContext dslContext, AclDeps aclDeps) {
+    this.dslContext = dslContext;
+    this.clientDAO = aclDeps.createClientDAO();
+    this.groupDAO = aclDeps.createGroupDAO();
+    this.secretContentDAO = aclDeps.createSecretContentDAO();
+    this.secretSeriesDAO = aclDeps.createSecretSeriesDAO();
```

Comment:
I think you should defer creating these objects until used because they will all work on the same DB handle.